### PR TITLE
Validate presence of ProgramConclusion on DegreeCurricularPlan approval ACDM-1051 #resolve

### PR DESCRIPTION
**Pull request by gil-l**

### CODE REVIEW

File: src/main/java/org/fenixedu/academic/domain/DegreeCurricularPlan.java [hunk 1]
```diff
@@ -113,7 +114,7 @@ public int compare(DegreeCurricularPlan o1, DegreeCurricularPlan o2) {
      * degree type 2. Reverse order of ExecutionDegrees 3. It's degree code (in
      * order to roughly order them by prebolonha/bolonha) OR reverse order of
      * their own name
-     *
```
Comment (pedrosan7os) [inline src/main/java/org/fenixedu/academic/domain/DegreeCurricularPlan.java]: avoid space commits

File: src/main/java/org/fenixedu/academic/domain/DegreeCurricularPlan.java [hunk 2]
```diff
@@ -956,7 +962,7 @@ public void doForAllCurricularCourses(final CurricularCourseFunctor curricularCo
     /**
      * Method to get a filtered list of a dcp's curricular courses, with at
      * least one open context in the given execution year
-     *
```
Comment (pedrosan7os) [inline src/main/java/org/fenixedu/academic/domain/DegreeCurricularPlan.java]: avoid space commits

File: src/main/java/org/fenixedu/academic/domain/DegreeCurricularPlan.java [hunk 3]
```diff
@@ -973,9 +979,9 @@ public void applyToCurricularCourses(final ExecutionYear executionYear, final Pr
 
     /**
      * Method to get an unfiltered list of a bolonha dcp's competence courses
-     *
```
Comment (pedrosan7os) [inline src/main/java/org/fenixedu/academic/domain/DegreeCurricularPlan.java]: avoid space commits

File: src/main/java/org/fenixedu/academic/domain/DegreeCurricularPlan.java [hunk 4]
```diff
@@ -990,7 +996,7 @@ public void applyToCurricularCourses(final ExecutionYear executionYear, final Pr
      * Method to get a filtered list of a dcp's competence courses in the given
      * execution year. Each competence courses is connected with a curricular
      * course with at least one open context in the execution year
-     *
```
Comment (pedrosan7os) [inline src/main/java/org/fenixedu/academic/domain/DegreeCurricularPlan.java]: avoid space commits

**Comment by gil-l** [comment]
Forgot the trimming in save actions, my bad, it's fixed now.

File: src/main/resources/resources/DomainExceptionResources_pt.properties [hunk 5]
```diff
@@ -45,6 +45,7 @@ error.degree.has.degree.curricular.plans = O curso possui planos curriculares as
 error.degreeCurricularPlan.already.approved = N�o pode aprovar o plano curricular, pois este j� esteve aprovado. Por favor contacte o suporte.
 error.degreeCurricularPlan.degreeStructure.cannot.be.changed.when.already.has.curricular.courses=N�o � poss�vel alterar a dura��o do plano curricular quando j� existem unidades curriculares associadas
 error.degreeCurricularPlan.existing.name.and.degree = J� existe um plano curricular com esse nome para o curso pretendido.
+error.degreeCurricularPlan.missing.program.conclusion = N�o � poss�vel aprovar o plano curricular pois este n�o inclui uma conclus�o de programa.
```
Comment (sergiofbsilva) [inline src/main/resources/resources/DomainExceptionResources_pt.properties]: program conclusion should be translated to "tipo de apuramento"

**Pull request merged**